File: adventofcode-2016/src/main/java/nl/mvdr/adventofcode/adventofcode2016/day16/Part2.java
package nl.mvdr.adventofcode.adventofcode2016.day16;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import nl.mvdr.adventofcode.solver.Solver;

public class Part2 implements Solver {

    private static final Logger LOGGER = LoggerFactory.getLogger(Part2.class);

    @Override
    public String solve(String inputfile) {
        return new DragonChecksumSolver(35651584).solve(inputfile);
    }

    public static void main(String[] args) {
        Part2 instance = new Part2();

        String result = instance.solve("input-day16.txt");

        LOGGER.info(result);
    }
}

File: adventofcode-2025/src/main/kotlin/nl/mvdr/adventofcode/adventofcode2025/day08/Part2.kt
package nl.mvdr.adventofcode.adventofcode2025.day08

import io.github.oshai.kotlinlogging.KotlinLogging
import nl.mvdr.adventofcode.solver.FunctionSolver

private val logger = KotlinLogging.logger{}

fun solvePart2(lines: Sequence<String>): Long {
    val junctionBoxes = parseJunctionBoxes(lines)
    val pairs = findPairs(junctionBoxes)
    val lastPair = findCircuits(junctionBoxes, pairs).lastPair
    return lastPair.first().x.toLong() * lastPair.last().x.toLong()
}

fun main() {
    val result = FunctionSolver(::solvePart2).solve("input-day08.txt")
    logger.info { result }
}
